### PR TITLE
Add a list for folders which should not be count in

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Bugfix: don't count non-user folder in occ user:report command

Before this PR several folders for example 'avatars' has been counted as user
folder via the occ user:report command.
With this PR a list of folders has been added which should be not count in,
serving a proper value of the user directory count.

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JanAckermann** [comment]
FYI @NannaBarz

File: core/Command/User/Report.php [hunk 1]
```diff
@@ -40,6 +41,14 @@ class Report extends Command {
 	/** @var UserTypeHelper */
 	protected $userTypeHelper;
 
+	/** @var array */
+	private $directoryIngoreList = [
```
Comment (phil-davis) [inline core/Command/User/Report.php:45]:
```suggestion
	private $directoryIgnoreList = [
```

File: changelog/unreleased/39216 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Bugfix: don't count non-user folder in occ user:report command
+
+Before this PR several folders for example 'avatars' has been counted as user
+folder via the occ user:report command.
+With this PR a list of folders has been added which should be not count in,
+serving a proper value of the user directory count.
```
Comment (phil-davis) [inline changelog/unreleased/39216:6]:
```suggestion
Before this PR several folders, for example 'avatars', have been counted as user
folders via the occ user:report command.
With this PR a list of folders has been added which should not be counted as user folders.
The user directory count is now correct.
```

File: core/Command/User/Report.php (hunk 1 above)
Comment (phil-davis) [reply core/Command/User/Report.php:45]: Should we put this somewhere common, that can be used to add to `$invalidUids` in https://github.com/owncloud/core/blob/master/lib/private/User/Manager.php ?

PR #39206 touched that last week.

**Review by phil-davis — COMMENTED** [review]
And there should  be a unit test of `countUserDirectories` that needs adjusting or adding to.

File: core/Command/User/Report.php [hunk 3]
```diff
@@ -99,7 +108,12 @@ private function countUsers() {
 
 	private function countUserDirectories() {
 		$dataview = new \OC\Files\View('/');
-		$userDirectories = $dataview->getDirectoryContent('/', 'httpd/unix-directory');
+		$directories = $dataview->getDirectoryContent('/', 'httpd/unix-directory');
+
+		$userDirectories = array_filter($directories, function (FileInfo $directory) {
```
Comment (JammingBen) [inline core/Command/User/Report.php:113]:
```suggestion
		$userDirectories = \array_filter($directories, function (FileInfo $directory) {
```

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIComments-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32281/124/1

File: core/Command/User/Report.php [hunk 4]
```diff
@@ -99,7 +108,12 @@ private function countUsers() {
...
+			return !in_array($directory->getName(), $this->directoryIngoreList);
```
Comment (JammingBen) [inline core/Command/User/Report.php]:
```suggestion
			return !\in_array($directory->getName(), $this->directoryIngoreList);
```

**Review by JammingBen — CHANGES_REQUESTED** [review]

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/32281/124/15
`element click intercepted: Element <a class="action action-share permanent" href="#" data-action="Share" data-original-title="" title="">...</a> is not clickable at point (767, 165). Other element would receive the click: <a href="#">...</a>` 

That seems to be an annoying "random" failure that has started happening: #39213 
It is unrelated to this PR. Someone will look into it.


I didn't restart CI, because there will be adjustments to the code of this PR  anyway.

File: core/Command/User/Report.php (hunk 1 above)
Comment (JanAckermann) [reply core/Command/User/Report.php:45]: I am thinking about lib/public/User/Constants.php @JammingBen @phil-davis 
searching for a good var name

Comment (phil-davis) [reply core/Command/User/Report.php:45]: Naming is hard. Maybe:

`public const DIRECTORIES_THAT_ARE_NOT_USERS`

And even put the files there as well?
`public const FILES_THAT_ARE_NOT_USERS`

And write a comment like "These files and directories can exist in the data directory along with user folders, and are not valid usernames"

**Review by phil-davis — APPROVED** [review]
Code and changelog LGTM. Let's see what happens to the tests...

**Comment by JanAckermann** [comment]
@pmaier1 FYI

**Review by JammingBen — APPROVED** [review]
Nice 🚀 

On a side note: In the future it would be helpful if apps could extend these arrays. Spaces for example introduces a new folder `files_spaces`, which we might want to exclude here, too.

**Comment by phil-davis** [comment]
@JanAckermann squash the commits when we know that tests are OK?

File: tests/Core/Command/User/ReportTest.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+use OC\Files\View;
+use OC\Helper\UserTypeHelper;
+use OC\User\User;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+use Test\TestCase;
+use Test\Traits\UserTrait;
+
+/**
+ * Class AddTest
```
Comment (phil-davis) [inline tests/Core/Command/User/ReportTest.php:36]:
```suggestion
 * Class ReportTest
```

**Review by phil-davis — APPROVED** [review]

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIComments-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32294/124/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIComments-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32297/124/1

**Comment by JanAckermann** [comment]
:face_with_head_bandage:

**Comment by phil-davis** [comment]
Issue #39213 is being a pain today!!! I will make a PR  to skip that new scenario while we try to make it reliable.

**Comment by JanAckermann** [comment]
Something fucked up the git history closing in favor of #39223

**Pull request closed without merging**